### PR TITLE
SpringData +Mvc

### PR DESCRIPTION
**Pull request by YevhenDov**

### CODE REVIEW

File: SpringDataMvc/src/main/java/com/company/controller/CrudController.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.springframework.web.servlet.ModelAndView;
+
+import java.util.List;
+import java.util.Map;
+
+@Controller
+@ComponentScan(basePackages = "com.company")
+public class CrudController {
+    @Autowired
+    UserService service;
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/controller/CrudController.java]: 1.What about encapsulation?
2.It is better to use a constructor for dependency injection.
https://www.vojtechruzicka.com/field-dependency-injection-considered-harmful/

File: SpringDataMvc/src/main/java/com/company/controller/CrudController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.ModelAndView;
+
+import java.util.List;
+import java.util.Map;
+
+@Controller
+@ComponentScan(basePackages = "com.company")
+public class CrudController {
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/controller/CrudController.java]: Think more about the controller name.

File: SpringDataMvc/src/main/java/com/company/controller/CrudController.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        service.createUser(user);
+
+        return "redirect:/";
+    }
+
+    @RequestMapping("/edit")
+    public ModelAndView editUserForm(@RequestParam int id) {
+        ModelAndView modelAndView = new ModelAndView("edit_user");
+        User user = service.getUserById(id);
+        modelAndView.addObject("user", user);
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/controller/CrudController.java]: add a blank line between lines 48 and 49

File: SpringDataMvc/src/main/java/com/company/controller/CrudController.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.List;
+import java.util.Map;
+
+@Controller
+@ComponentScan(basePackages = "com.company")
+public class CrudController {
+    @Autowired
+    UserService service;
+
+    @RequestMapping("/")
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/controller/CrudController.java]: Please look at annotations: 
@GetMapping
@PostMapping
@PutMapping
@DeleteMapping
@PatchMapping
https://www.baeldung.com/spring-requestmapping

File: SpringDataMvc/src/main/java/com/company/controller/CrudController.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    @RequestMapping("/edit")
+    public ModelAndView editUserForm(@RequestParam int id) {
+        ModelAndView modelAndView = new ModelAndView("edit_user");
+        User user = service.getUserById(id);
+        modelAndView.addObject("user", user);
+        return modelAndView;
+    }
+
+    @RequestMapping("/delete")
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/controller/CrudController.java]: Why this method is GET but not DELETE?

File: SpringDataMvc/src/main/java/com/company/listener/CreatableEnmityListener.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import com.company.entity.CreatableEntity;
+
+import javax.persistence.PrePersist;
+import java.util.Date;
+
+public class CreatableEnmityListener {
+
+    @PrePersist
+    public void prePersist(CreatableEntity entity){
+        entity.setCreatedDate(new Date().getTime());
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/listener/CreatableEnmityListener.java]: Why not LocalDateTime.now()?

File: SpringDataMvc/src/main/java/com/company/controller/CrudController.java (hunk 1 above)
Comment (YevhenDov) [reply SpringDataMvc/src/main/java/com/company/controller/CrudController.java]: The class has one var, why would I need a constructor for injection??

File: SpringDataMvc/src/main/java/com/company/controller/CrudController.java (hunk 5 above)
Comment (YevhenDov) [reply SpringDataMvc/src/main/java/com/company/controller/CrudController.java]: Because I need get an id for a delete.

File: SpringDataMvc/mvnw [hunk 7]
```diff
@@ -0,0 +1,286 @@
+#!/bin/sh
```
Comment (Anderwerty) [inline SpringDataMvc/mvnw]: Should this file in a git repository?

File: SpringDataMvc/mvnw.cmd [hunk 8]
```diff
@@ -0,0 +1,161 @@
+@REM ----------------------------------------------------------------------------
```
Comment (Anderwerty) [inline SpringDataMvc/mvnw.cmd]: Should this file in a git repository?

File: SpringDataMvc/src/main/java/com/company/transformer/UserTransformer.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserTransformer {
+
+    @Autowired
+    private UserEntityRepository repository;
+
+    private UserEntity userEntity;
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/transformer/UserTransformer.java]: A global variable is not good practice. Please remove it.

File: SpringDataMvc/mvnw (hunk 7 above)
Comment (Anderwerty) [reply SpringDataMvc/mvnw]: Should this file in a git repository?

File: SpringDataMvc/mvnw.cmd (hunk 8 above)
Comment (Anderwerty) [reply SpringDataMvc/mvnw.cmd]: Should this file in a git repository?

File: .gitignore [hunk 10]
```diff
@@ -1,4 +1,5 @@
 *.iml
 .idea
+.mvn
 /out
 client_secrets.json
```
Comment (Anderwerty) [inline .gitignore]: Add a blank line at the end of the file.

File: SpringDataMvc/src/main/java/com/company/controller/UserController.java [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @PostMapping("/addOrUpdate")
+    public String addOrUpdate(@ModelAttribute("user") User user) {
+        service.createUser(user);
+
+        return "redirect:/";
+    }
+
+    @GetMapping("/edit")
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/controller/UserController.java]: Where is request parameter?

File: SpringDataMvc/src/main/java/com/company/controller/UserController.java [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+    public ModelAndView editUserForm(@RequestParam int id) {
+        ModelAndView modelAndView = new ModelAndView("edit_user");
+        User user = service.getUserById(id);
+        modelAndView.addObject("user", user);
+
+        return modelAndView;
+    }
+
+    @GetMapping("/delete")
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/controller/UserController.java]: Where is request parameter?

File: SpringDataMvc/src/main/java/com/company/controller/UserController.java (hunk 11 above)
Comment (YevhenDov) [reply SpringDataMvc/src/main/java/com/company/controller/UserController.java]: view my index.jsp

File: SpringDataMvc/src/main/java/com/company/controller/UserController.java (hunk 12 above)
Comment (YevhenDov) [reply SpringDataMvc/src/main/java/com/company/controller/UserController.java]: view my index.jsp

Comment (YevhenDov) [reply SpringDataMvc/src/main/java/com/company/controller/UserController.java]: > view my index.jsp
`<a href="edit?id=${user.id}">Edit</a>`

File: SpringDataMvc/src/main/java/com/company/entity/UserEntity.java [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+    @Column(name = "username")
+    private String username;
+
+    @Column(name = "password")
+    private String password;
+
+    @Transient
+    transient private String confirmPassword;
+
+//    @ManyToMany
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/entity/UserEntity.java]: Do you need this code?

File: SpringDataMvc/src/main/java/com/company/service/impl/UserServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    @Override
+    public void deleteUserById(Integer id) {
+        repository.deleteById(id);
+    }
+
+    @Override
+    public List<User> getAllUsers() {
+        List<UserEntity> allUserEntity = repository.findAll();
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/service/impl/UserServiceImpl.java]: You could use the stream API here.

File: SpringDataMvc/src/main/java/com/company/transformer/UserTransformer.java [hunk 15]
```diff
@@ -0,0 +1,32 @@
+package com.company.transformer;
+
+import com.company.dto.User;
+import com.company.entity.UserEntity;
+import com.company.repository.UserEntityRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserTransformer {
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/transformer/UserTransformer.java]: UserTransformer -> UserMapper

File: SpringDataMvc/src/main/java/com/company/transformer/UserTransformer.java [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+    public UserEntity buildEntity(User user) {
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/transformer/UserTransformer.java]: buildEntity -> mapUserToUserEntity

File: SpringDataMvc/src/main/java/com/company/transformer/UserTransformer.java [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+
+        userEntity.setId(user.getId());
+        userEntity.setName(user.getName());
+        userEntity.setAge(user.getAge());
+        userEntity.setEmail(user.getEmail());
+
+        return userEntity;
+    }
+
+    public User buildUser(UserEntity userEntity) {
```
Comment (Anderwerty) [inline SpringDataMvc/src/main/java/com/company/transformer/UserTransformer.java]: buildUser ->mapUserEntityToUser

File: SpringDataMvc/src/main/java/com/company/entity/UserEntity.java (hunk 13 above)
Comment (YevhenDov) [reply SpringDataMvc/src/main/java/com/company/entity/UserEntity.java]: No it for next homework

File: SpringDataMvc/src/main/java/com/company/service/impl/UserServiceImpl.java (hunk 14 above)
Comment (YevhenDov) [reply SpringDataMvc/src/main/java/com/company/service/impl/UserServiceImpl.java]: I know,  but foreach simpler for me

File: SpringDataMvc/src/main/java/com/company/entity/UserEntity.java (hunk 13 above)
Comment (Anderwerty) [reply SpringDataMvc/src/main/java/com/company/entity/UserEntity.java]: Better remove it for this PR.

**Pull request closed without merging**